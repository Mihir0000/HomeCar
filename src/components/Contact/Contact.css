.contact_section {
    width: 100%;
    float: left;
    background-image: url(../../../public/static/img/contact-bg.png);
    background-size: 100%;
    background-repeat: no-repeat;
    padding-bottom: 150px;
}


.layout_padding {
    padding-top: 90px;
    padding-bottom: 0px;
}

.contact_taital {
    width: 100%;
    float: left;
    font-size: 50px;
    color: #030202;
    font-weight: bold;
}

.news_section_2 {
    width: 100%;
    float: left;
}

.icon_main {
    width: 100%;
    display: flex;
    padding-top: 50px;
}

.icon_7 {
    width: 20%;
    float: left;
}

.diabetes_text {
    width: 70%;
    float: left;
    font-size: 30px;
    color: #040101;
    font-weight: 500;
}

.contact_box {
    width: 100%;
    float: left;
    background-color: #f1fafb;
    height: auto;
    padding: 30px;
}

.book_text {
    width: 100%;
    float: left;
    font-size: 30px;
    color: #252525;
    text-align: center;
    font-weight: 500;
}
.Email_text {
    width: 100%;
    float: left;
    font-size: 18px;
    color: #bebebe;
    background-color: #ffffff;
    padding: 5px 15px;
    border: 0px;
    margin-top: 30px;
}

.massage-bt {
    width: 100%;
    float: left;
    color: #bebebe;
    height: 110px;
    font-size: 18px;
    padding: 36px 20px 10px 20px;
    background-color: #ffffff;
    border: 0px;
    margin-top: 30px;
}

.send_bt {
    width: 140px;
    margin: 0 auto;
    text-align: center;
}

.send_bt a {
    width: 100%;
    float: left;
    font-size: 18px;
    color: #ffffff;
    background-color: #1becde;
    padding: 7px 0px;
    text-align: center;
    margin-top: 50px;
    text-transform: uppercase;
}

.send_bt a:hover {
    background-color: #252525;
}