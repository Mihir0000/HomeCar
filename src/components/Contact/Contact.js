import React from 'react';
import { Card, Col, Container, Form, InputGroup, Row } from 'react-bootstrap';
import './Contact.css';

const Contact = ({ author }) => {
    return (
        <Container fluid className="contact_section layout_padding">
            <Container>
                <h1 className="contact_taital">What we do, {author}</h1>
                <div className="news_section_2">
                    <Row>
                        <Col md={6}>
                            <Card
                                className="icon_main bg-transparent flex-row"
                                style={{ border: 'none' }}
                            >
                                <Card.Body className="icon_7 p-0">
                                    <Card.Img
                                        style={{ width: '40%' }}
                                        src="static/img/icon-7.png"
                                        alt="icon"
                                    />
                                </Card.Body>
                                <Card.Title className="diabetes_text fs-2">
                                    Diabetes and obesity Counselling{' '}
                                </Card.Title>
                            </Card>
                            <Card
                                className="icon_main bg-transparent flex-row"
                                style={{ border: 'none' }}
                            >
                                <Card.Body className="icon_7">
                                    <Card.Img
                                        style={{ width: '40%' }}
                                        src="static/img/icon-5.png"
                                        alt="icon"
                                    />
                                </Card.Body>
                                <Card.Title className="diabetes_text fs-2">
                                    Obstetrics and Gynsecology
                                </Card.Title>
                            </Card>
                            <Card
                                className="icon_main bg-transparent flex-row"
                                style={{ border: 'none' }}
                            >
                                <Card.Body className="icon_7">
                                    <Card.Img
                                        style={{ width: '40%' }}
                                        src="static/img/icon-6.png"
                                        alt="icon"
                                    />
                                </Card.Body>
                                <Card.Title className="diabetes_text fs-2">
                                    Surgical and medical Oncology
                                </Card.Title>
                            </Card>
                        </Col>
                        <Col md={6}>
                            <InputGroup className="contact_box d-flex flex-column">
                                <InputGroup.Text className="book_text fs-2 justify-content-center bg-transparent border-0">
                                    Book Appoinment
                                </InputGroup.Text>
                                <Card
                                    className="bg-transparent"
                                    style={{ border: 'none' }}
                                >
                                    <Form.Control
                                        className="Email_text"
                                        placeholder="Name"
                                        aria-label="Name"
                                        aria-describedby="basic-addon2"
                                    />
                                    <Form.Control
                                        className="Email_text"
                                        placeholder="Email"
                                        aria-label="Email"
                                        aria-describedby="basic-addon2"
                                    />
                                </Card>
                                <Form.Control
                                    className="massage-bt"
                                    rows="5"
                                    id="comment"
                                    name="Massage"
                                    as="textarea"
                                    placeholder="Leave a comment here"
                                    style={{ height: '100px', width: '100%' }}
                                />
                                <Card
                                    className="send_bt bg-transparent"
                                    style={{ border: 'none' }}
                                >
                                    <a href="/#">SEND</a>
                                </Card>
                            </InputGroup>
                        </Col>
                    </Row>
                </div>
            </Container>
        </Container>
    );
};

export default Contact;
