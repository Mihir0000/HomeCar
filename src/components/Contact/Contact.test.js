import React from 'react';
import { render, screen } from '@testing-library/react';
import Contact from './Contact';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

test('Book appoinment name should be rendered', () => {
    render(<Contact />);
    const inputName = screen.getByPlaceholderText(/name/i);
    expect(inputName).toBeInTheDocument();
});

test('Book appoinment email value should be empty first time', () => {
    render(<Contact />);
    const inputEmail = screen.getByPlaceholderText(/Email/i);
    expect(inputEmail.value).toBe('');
});

test('Button input should be rendered', () => {
    render(<Contact />);
    const buttonInput = screen.getByTestId('contact-btn');
    expect(buttonInput).toBeInTheDocument();
});

test('snapshot testing --Contact', () => {
    const component = renderer.create(<Contact />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

// crash test --Contact
it('render without crashing --contact', () => {
    const div = document.createElement('div');
    render(<Contact />, div);
    ReactDOM.unmountComponentAtNode(div);
});