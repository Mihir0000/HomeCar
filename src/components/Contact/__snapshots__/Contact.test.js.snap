// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing --Contact 1`] = `
<div
  className="contact_section layout_padding container-fluid"
>
  <div
    className="container"
  >
    <p
      className="contact_taital card-text"
    >
      What we do, 
    </p>
    <div
      className="news_section_2 container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <div
            className="icon_main bg-transparent flex-row card"
            style={
              Object {
                "border": "none",
              }
            }
          >
            <div
              className="icon_7 p-0 card-body"
            >
              <img
                alt="icon"
                className="card-img"
                src="static/img/icon-7.png"
                style={
                  Object {
                    "width": "40%",
                  }
                }
              />
            </div>
            <div
              className="diabetes_text fs-2 card-title h5"
            >
              Diabetes and obesity Counselling
               
            </div>
          </div>
          <div
            className="icon_main bg-transparent flex-row card"
            style={
              Object {
                "border": "none",
              }
            }
          >
            <div
              className="icon_7 card-body"
            >
              <img
                alt="icon"
                className="card-img"
                src="static/img/icon-5.png"
                style={
                  Object {
                    "width": "40%",
                  }
                }
              />
            </div>
            <div
              className="diabetes_text fs-2 card-title h5"
            >
              Obstetrics and Gynsecology
            </div>
          </div>
          <div
            className="icon_main bg-transparent flex-row card"
            style={
              Object {
                "border": "none",
              }
            }
          >
            <div
              className="icon_7 card-body"
            >
              <img
                alt="icon"
                className="card-img"
                src="static/img/icon-6.png"
                style={
                  Object {
                    "width": "40%",
                  }
                }
              />
            </div>
            <div
              className="diabetes_text fs-2 card-title h5"
            >
              Surgical and medical Oncology
            </div>
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div
            className="contact_box d-flex flex-column input-group"
          >
            <span
              className="book_text fs-2 justify-content-center bg-transparent border-0 input-group-text"
            >
              Book Appoinment
            </span>
            <div
              className="bg-transparent card"
              style={
                Object {
                  "border": "none",
                }
              }
            >
              <input
                aria-describedby="basic-addon2"
                aria-label="Name"
                className="Email_text form-control"
                placeholder="Name"
              />
              <input
                aria-describedby="basic-addon2"
                aria-label="Email"
                className="Email_text form-control"
                placeholder="Email"
              />
            </div>
            <textarea
              className="massage-bt form-control"
              id="comment"
              name="Massage"
              placeholder="Leave a comment here"
              rows="5"
              style={
                Object {
                  "height": "100px",
                  "width": "100%",
                }
              }
            />
            <div
              className="send_bt bg-transparent card"
              data-testid="contact-btn"
              style={
                Object {
                  "border": "none",
                }
              }
            >
              <a
                className="card-link"
                href="/#"
              >
                SEND
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
